lightwood/encoders/time_series/rnn.py

```diff
@@ -149,26 +197,76 @@ def encode(self, column_data, get_next_count = None):
         else:
             return self._pytorch_wrapper(torch.stack(ret)), self._pytorch_wrapper(torch.stack(next))
 
+    def _decode_one(self, hidden):
+        """
+        Decodes a single time series from its encoded representation.
+        :param hidden: time series embedded representation tensor, with size self._encoded_vector_size
+        :return: decoded time series list
+        """
+        self._decoder.eval()
+        with torch.no_grad():
+            ret = []
+            next_tensor = torch.full((1, 1, self._n_dims), self._sos, dtype=torch.float32).to(self.device)
+            for _ in range(self._max_dec_length):
+                next_tensor, hidden = self._decoder.forward(next_tensor, hidden)
+                ret.append(next_tensor)
+            return torch.stack(ret)
+
+    def decode(self, encoded_data):
+        """
+        Decode a list of embedded multidimensional time series
+        :param encoded_data: a list of embeddings [ e1, e2, ...] to be decoded into time series
+        :return: a list of reconstructed time series
+        """
+        if not self._prepared:
+            raise Exception('You need to call "prepare_encoder" before calling "encode" or "decode".')
 
+        ret = []
+        for _, val in enumerate(encoded_data):
+            hidden = torch.unsqueeze(torch.unsqueeze(val, dim=0), dim=0).to(self.device)
+            reconstruction = self._decode_one(hidden).cpu().squeeze().T.tolist()
+            ret += [reconstruction]
 
-# only run the test if this file is called from debugger
-if __name__ == "__main__":
-    series = []
-    for j in range(100):
-        length = 3 + np.random.randint(14)
-        skip = 1+np.random.randint(4)
-
-        start = np.random.randint(30)
-        vec = [start+j*skip for j in range(length)]
-
-        series+=[' '.join([str(x) for x in vec])]
-
-    encoder = RnnEncoder(encoded_vector_size=3,train_iters=10)
-    encoder.prepare_encoder(series, feedback_hoop_function=lambda x:print(x))
-
+        return self._pytorch_wrapper(ret)
 
-    # test de decoder
-    init_vector = ['31 33 35 37', '1 2 3 4 5 6']
 
-    print(encoder.encode(column_data=init_vector, get_next_count=2))
-    print(encoder.encode(column_data=init_vector))
+# only run the test if this file is called from debugger
+if __name__ == "__main__":
+    from lightwood.encoders.time_series.helpers.test_ts import test_ts
+
+    # set log, fix random seed for reproducibility
+    logging.basicConfig(level=logging.DEBUG)
+    seed = 42
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    if 'cuda' in str(get_devices()[0]):
+        torch.backends.cudnn.deterministic = True
+        torch.backends.cudnn.benchmark = False
```

Comment:
Hmh, I guess this code is needed here but it shouldn't be.

Currently this already happens in `default_net.py`, it should instead happen in `__init__.py` so that we don't have to manually do it in every test. Can you remove it from here and add it as a snipet in `__init__.py` (I guess keep the seeding the `default_net.py` as well, since otherwise we'd be having the same feed for each run of lightwood, and that might bias us in a weird direction (?) maybe not (?)